src/miniscript/types/extra_props.rs

```diff
@@ -8,6 +8,90 @@ use MiniscriptKey;
 use Terminal;
 
 pub const MAX_OPS_PER_SCRIPT: usize = 201;
+// https://github.com/bitcoin/bitcoin/blob/9ccaee1d5e2e4b79b0a7c29aadb41b97e4741332/src/script/script.h#L39
+pub const HEIGHT_TIME_THRESHOLD: u32 = 500_000_000;
+
+/// Helper struct Whether any satisfaction of this fragment contains any timelocks
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Debug, Hash)]
+pub struct TimeLockInfo {
+    // csv with heights
+    pub csv_with_height: bool,
+    /// csv with times
+    pub csv_with_time: bool,
+    /// cltv with heights
+    pub cltv_with_height: bool,
+    /// cltv with times
+    pub cltv_with_time: bool,
+    /// combination of any heightlocks and timelocks
+    pub contains_combination: bool,
```

Comment:
Do we actually need this field? Or can we just check `(csv_with_height && csv_with_time) || (cltv_with_height && cltv_with_time)`?

Reply:
Ah - it is there to cover the disjunction case, where you may have a heighlock branch and a timelock branch, but they are independent so it is ok.